plugins/lsp/language-servers/_mk-lsp.nix

```diff
@@ -7,6 +7,8 @@
 { pkgs, ... }:
 {
   name,
+  aliases ? [ ],
+  oldNames ? [ ],
```

Comment:
Do you _want_ support for both aliases (no-warning) and renames (warnings), or would you rather we only did renames with warnings?

Reply:
Right now you have them both enabled ? What's the behavior ?

Reply:
I have both _available_, but only `aliases` is in use. Even that is only used indirectly, via `serverName` automatically getting an alias.

This means users are free to use `name` and `serverName` interchangeably without a warning (docs will still show `name` of course).

Having the `alias` and/or `oldNames` arguments available means a server _could_ explicitly specify additional aliases and/or renames, however this is currently unused and may be overkill.

The automatic `serverName` alias could also be switched to a rename, if a warning is desired.

Reply:
All right. I would go for a rename then. We eventually want to get rid of the rename/warning and maintain a single name.

Reply:
> All right. I would go for a rename then.

Fine with me. Should I simplify `mkLsp` to _only_ do this for the `serverName` case, or should the `alias`/`oldNames` arguments stay for future use?

> We eventually want to get rid of the rename/warning and maintain a single name.

Personally, I don't see the harm in maintaining the rename indefinitely - especially if it's automatic. New users _guessing_ the module name based on lspconfig docs instead of ours may appreciate the warning?

I guess we could replace the warning with an assertion (in the future), but that'd mean implementing our own custom version of `mkRenamedOptionModule`.

Reply:
> Fine with me. Should I simplify mkLsp to only do this for the serverName case, or should the alias/oldNames arguments stay for future use?

I would drop the alias code if it is unused, yes.

> Personally, I don't see the harm in maintaining the rename indefinitely - especially if it's automatic. New users guessing the module name based on lspconfig docs instead of ours may appreciate the warning?

Just leave a TODO for us not to forget that it is there, but let's keep it at least until 24.11.

Reply:
Done